hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationExecutorImpl.java

```diff
@@ -112,17 +118,19 @@ public OperationExecutorImpl(GroupProperties properties,
     }
 
     private OperationRunner[] initGenericOperationRunners(GroupProperties properties, OperationRunnerFactory runnerFactory) {
-        int genericThreadCount = properties.getInteger(GENERIC_OPERATION_THREAD_COUNT);
-        if (genericThreadCount <= 0) {
+        int threadCount = properties.getInteger(GENERIC_OPERATION_THREAD_COUNT);
+        if (threadCount <= 0) {
             // default generic operation thread count
             int coreSize = Runtime.getRuntime().availableProcessors();
-            genericThreadCount = Math.max(2, coreSize / 2);
+            threadCount = Math.max(2, coreSize / 2);
         }
 
-        OperationRunner[] operationRunners = new OperationRunner[genericThreadCount];
+
+        OperationRunner[] operationRunners = new OperationRunner[threadCount + priorityThreadCount];
```

Comment:
Good one.

However I would like this to get solved on the property. So when you define a property, you should be able to set a validator on it. In a lot of cases, the same property is being pulled out many times; e.g. partition count. Where should this property be validated?